Sources/NIO/ConvenienceOptionSupport.swift

```diff
@@ -0,0 +1,186 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// MARK:  Universal Client Bootstrap
+extension NIOClientTCPBootstrapProtocol {
+    /// Apply any understood convenience options to the bootstrap, removing them from the set of options if they are consumed.
+    /// - parameters:
+    ///     - options:  The options to try applying - the options applied should be consumed from here.
+    /// - returns: The updated bootstrap with and options applied.
+    public func _applyChannelConvenienceOptions(_ options: inout ChannelOptions.TCPConvenienceOptions) -> Self {
+        // Default is to consume no options and not update self.
+        return self
+    }
+}
+
+extension NIOClientTCPBootstrap {
+    /// Specifies some `TCPConvenienceOption`s to be applied to the channel.
+    /// These are preferred over regular channel options as they are easier to use and restrict
+    /// options to those with a normal user would consider.
+    /// - Parameter options: Set of convenience options to apply.
+    /// - Returns: The updated bootstrap (`self` being mutated)
+    public func channelConvenienceOptions(_ options: ChannelOptions.TCPConvenienceOptions) -> NIOClientTCPBootstrap {
+        var optionsRemaining = options
+        // First give the underlying a chance to consume options.
+        let withUnderlyingOverrides =
+            NIOClientTCPBootstrap(self, withUpdated:
+                                    underlyingBootstrap._applyChannelConvenienceOptions(&optionsRemaining))
+        // Default apply any remaining options.
+        return optionsRemaining.applyFallbackMapping(withUnderlyingOverrides)
+    }
+}
+
+// MARK: Utility
+extension ChannelOptions.Types {
+    /// Has an option been set?
+    /// Option has a value of generic type ValueType.
+    public enum ConvenienceOptionValue<ValueType> {
+        /// The option was not set.
+        case notSet
+        /// The option was set with a value of type ValueType.
+        case set(ValueType)
+    }
+}
+
+extension ChannelOptions.Types.ConvenienceOptionValue where ValueType == () {
+    /// Convenience method working with bool options as bool values for set.
+    public var isSet: Bool {
+        get {
+            switch self {
+            case .notSet:
+                return false
+            case .set(()):
+                return true
+            }
+        }
+    }
+}
+
+extension ChannelOptions.Types.ConvenienceOptionValue where ValueType == () {
+    fileprivate init(flag: Bool) {
```

Comment:
[just in case you're making more changes anyway], I'd remove the `flag:` label, ie `_ flag: Bool`. It doesn't add value

Reply:
It distinguishes from having ValueType == Bool.